tests/Unit/ApparentHorizons/Test_StrahlkorperDataBox.cpp

```diff
@@ -0,0 +1,285 @@
+
+    test_dx_radius.get(0)[s] = -((cos_phi * sin_phi) / r) +
+                               (cos_phi * square(cos_theta) * sin_phi) / r;
+    test_dx_radius.get(1)[s] =
+        square(cos_phi) / r + (square(cos_theta) * square(sin_phi)) / r;
+    test_dx_radius.get(2)[s] = -((cos_theta * sin_phi * sin_theta) / r);
+    for (auto& a : test_dx_radius) {
+      a[s] *= amp;
+    }
+  }
+  const auto& strahlkorper_dx_radius =
+      db::get<StrahlkorperTags::DxRadius<Frame::Inertial>>(box);
+  CHECK_ITERABLE_APPROX(strahlkorper_dx_radius, test_dx_radius);
+
+  // Test second derivatives.
+  db::item_type<StrahlkorperTags::D2xRadius<Frame::Inertial>> test_d2x_radius(
+      n_pts);
+  for (size_t s = 0; s < n_pts; ++s) {
+    // Messy analytic solution Mark Scheel computed in Mathematica
+    const double theta = theta_phi[0][s];
+    const double phi = theta_phi[1][s];
+    const double r = test_radius[s];
+    const double sin_theta = sin(theta);
+    const double cos_theta = cos(theta);
+    const double sin_phi = sin(phi);
+    const double cos_phi = cos(phi);
+    const double cos_2_theta = cos(2 * theta);
+    const double amp = -sqrt(3.0 / 8.0 / M_PI) * y11_amplitude;
+    test_d2x_radius.get(0, 0)[s] =
+        (9 * sin(3 * phi) * sin_theta -
+         sin_phi * (7 * sin_theta + 12 * square(cos_phi) * sin(3 * theta))) /
+        (16. * square(r));
+    test_d2x_radius.get(0, 1)[s] =
+        -((cos_phi *
+           (square(cos_phi) + ((-1 + 3 * cos_2_theta) * square(sin_phi)) / 2.) *
+           sin_theta) /
+          square(r));
+    test_d2x_radius.get(0, 2)[s] =
+        (3 * cos_phi * cos_theta * sin_phi * square(sin_theta)) / square(r);
+    test_d2x_radius.get(1, 1)[s] =
+        (-3 * (7 * sin_phi * sin_theta + 3 * sin(3 * phi) * sin_theta +
+               4 * pow<3>(sin_phi) * sin(3 * theta))) /
+        (16. * square(r));
+    test_d2x_radius.get(1, 2)[s] =
+        -((cos_theta * (square(cos_phi) +
+                        ((-1 + 3 * cos_2_theta) * square(sin_phi)) / 2.)) /
+          square(r));
+    test_d2x_radius.get(2, 2)[s] =
+        ((1 + 3 * cos_2_theta) * sin_phi * sin_theta) / (2. * square(r));
+    for (auto& a : test_d2x_radius) {
+      a[s] *= amp;
+    }
+  }
+  const auto& strahlkorper_d2x_radius =
+      db::get<StrahlkorperTags::D2xRadius<Frame::Inertial>>(box);
+  CHECK_ITERABLE_APPROX(test_d2x_radius, strahlkorper_d2x_radius);
+
+  // Test nabla squared
+  DataVector test_laplacian(n_pts);
+  for (size_t s = 0; s < n_pts; ++s) {
+    func.scalar_laplacian(&test_laplacian, 1, s, {theta_phi[0][s]},
+                          {theta_phi[1][s]});
+    test_laplacian[s] *= y11_amplitude;
+  }
+  const auto& strahlkorper_laplacian =
+      db::get<StrahlkorperTags::LaplacianRadius<Frame::Inertial>>(box);
+  CHECK_ITERABLE_APPROX(strahlkorper_laplacian, test_laplacian);
+}
+
+void test_normals() {
+  // Create a strahlkorper with a Im(Y11) dependence.
```

Comment:
factor out the creation of a Im(Y11) strahlkorper as it is identical to the last function